infra/Pos/Slotting/Impl/Util.hs

```diff
@@ -18,54 +18,102 @@ import           Pos.Core.Types              (EpochIndex, SlotId (..), Timestamp
 import           Pos.Util.Util               (leftToPanic)
 
 import           Pos.Slotting.MemState.Class (MonadSlotsData (..))
-import           Pos.Slotting.Types          (EpochSlottingData (..), SlottingData (..))
+import           Pos.Slotting.Types          (EpochSlottingData (..))
 
 -- | Approximate current slot using outdated slotting data.
-approxSlotUsingOutdated :: MonadSlotsData m => EpochIndex -> Timestamp -> m SlotId
-approxSlotUsingOutdated penult t = do
-    SlottingData {..} <- getSlottingData
-    systemStart <- getSystemStart
-    let epochStart = esdStartDiff sdLast `addTimeDiffToTimestamp` systemStart
+approxSlotUsingOutdated
+    :: (MonadSlotsData m)
+    => Timestamp
+    -> m SlotId
+approxSlotUsingOutdated t = do
+
+    systemStart            <- getSystemStartM
+    (currentEpochIndex ,_) <- getCurrentNextEpochIndexM
+    (_, nextSlottingData)  <- getCurrentNextEpochSlottingDataM
```

Comment:
I know, it occurred to me. Seems like the only correct way is to use what you guys proposed - `(MonadReader ctx m, HasSlottingVar ctx)` or similar - a single source of information.
At this point, I think it wouldn't be a problem if I tried to change that, seems like a relatively small change to me. I could be wrong though.